### PR TITLE
Strg 2399 identify the element in the sdk

### PR DESCRIPTION
**Pull request by DvirMalka**
## Context

1. Fetch all of the user's assessments from strigo-app internal route (config)
2. Identify element with mutation observer on the correct window
3. send the custom event to strigo app

## Tests

- [x] Iframe widget
- [x] Overlay widget

### CODE REVIEW

File: dist/development/strigo.sdk.js [hunk 1]
```diff
@@ -244,6 +244,7 @@ ${JSON.stringify(parsedContext)}` : "");
         throw new Error(`Failed to fetch remote configuration: ${response.statusText}`);
       }
       const configuration = await response.json();
+      console.log(configuration);
```
Comment (guzzur) [inline dist/development/strigo.sdk.js:247]: Remove

File: src/modules/assessments-storage/assessments-storage.types.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export const enum INTRERCTION_TYPES {
```
Comment (guzzur) [inline src/modules/assessments-storage/assessments-storage.types.ts:1]:
```suggestion
export const enum INTERACTION_TYPES {
```

File: src/modules/assessments-storage/assessments-storage.types.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  CONTENT = 'content',
+  CLICK = 'click'
+}
+export interface StrigoAssessmentsStorage {
+  assessments: StrigoAssessment[];
+}
+
+export interface StrigoAssessment {
+  interactionEventType: 'content',
```
Comment (guzzur) [inline src/modules/assessments-storage/assessments-storage.types.ts:10]:
```suggestion
  interactionEventType: INTERACTION_TYPES.CONTENT,
```

File: src/modules/no-code-assessment/no-code-assessment.types.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+export const enum INTRERCTION_TYPES {
+    CONTENT = 'content',
+    CLICK = 'click'
+}
+  
+export interface NoCodeAssessment {
+    interactionEventType: 'content',
```
Comment (guzzur) [inline src/modules/no-code-assessment/no-code-assessment.types.ts:7]:
```suggestion
    interactionEventType: INTERACTION_TYPES.CONTENT,
```

File: src/modules/no-code-assessment/no-code-assessment.types.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+export const enum INTRERCTION_TYPES {
```
Comment (guzzur) [inline src/modules/no-code-assessment/no-code-assessment.types.ts:1]: These types are duplicated

File: src/modules/widgets/iframe.ts [hunk 6]
```diff
@@ -75,7 +75,7 @@ class IframeWidget implements IStrigoWidget {
       gutterSize: 2,
     });
 
-    this.initEventListeners(academyPlayerFrame, childFrame);
+    this.initEventListeners(academyPlayerFrame, childFrame);
```
Comment (guzzur) [inline src/modules/widgets/iframe.ts:78]:
```suggestion
    this.initEventListeners(academyPlayerFrame, childFrame);
```

**Review by guzzur — APPROVED** [review]

File: src/modules/widgets/overlay.ts [hunk 7]
```diff
@@ -28,6 +29,7 @@ class OverlayWidget implements IOverlayWidget {
     });
     const academyPlayerFrame = documentTools.createWidget(urlTools.generateStrigoIframeURL(configManager.getConfig()));
     this.initEventListeners(academyPlayerFrame);
+    noCodeAssessment.addDocumentObserver(window);
```
Comment (v0vs) [inline src/modules/widgets/overlay.ts:32]: I think we need to remove the observers on shutdown

Comment (DvirMalka) [reply src/modules/widgets/overlay.ts:32]: Will do

Comment (DvirMalka) [reply src/modules/widgets/overlay.ts:32]: https://github.com/strigo/strigo-sdk/pull/68

**Pull request merged**